ci/azure-pipelines-dev.yml

```diff
@@ -558,5 +558,8 @@ stages:
               versionSpec: '3.7'
             displayName: 'Use Python 3.7'
 
+          - script: ./ci/checks/check_min_airflow_dependency_compatibility.sh
+            name: CheckMinAirflowDependencyCompatibility.sh
+
```

Comment:
The goal here is a minimal test that block PR merges to develop if we have a fatal error during import time or when `get_context()` is called.

Reply:
I think this is good for now. I have been thinking as our pipelines get more complex we will need to use something like [pipeline triggers](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/pipeline-triggers?view=azure-devops) to DRY our pipelines more - e.g. have a pipeline that just does linting and have that as a prerequisite for other pipelines. Instead of including linting steps in all of our pipelines. So for example there would be an airflow compatibility pipeline where this would live (similarly sqlalchemy, pandas, spark etc compatibility pipelines) triggered after prereqs like linting.

Reply:
I wonder if we could use terraform instead of this and leverage its module composition functionality to DRY our pipelines while avoiding multiple pipelines and triggers.

Reply:
Interesting - that might also avoid lock in to azure.